scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -167,12 +167,12 @@ func (km *KafkaManager) loadOrStorePipeline(resourceName string, isModel bool) (
 		if err != nil {
 			return nil, err
 		}
-		pipeline.wg.Add(1) // wait set to allow consumer to say when started
 
 		val, loaded := km.pipelines.LoadOrStore(key, pipeline)
 		if loaded { // we can still have a race condition where multiple "create" are happening, we have to store the first one
 			pipeline = val.(*Pipeline)
 		} else {
+			pipeline.wg.Add(1) // wait set to allow consumer to say when started
```

Comment:
I think that there might be a race condition if put it here:
if we have 2 infer requests at the same time for a new pipeline: r1 and r2.
r1 is going to create the pipeline and r1 will get the reference at line 171. r2 is faster and will proceed until `pipeline.wg.Wait()` without waiting for r1 to start the consume at line 175.?